eq-author/src/components/CommentsPanel/index.js

```diff
@@ -0,0 +1,202 @@
+  width: 100%;
+  height: 0;
+  padding: 25% 0;
+`;
+
+const NameWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  -webkit-flex-direction: column;
+  justify-content: space-between;
+  align-items: start;
+  flex-grow: 2;
+`;
+
+const FlexLabel = styled.div`
+  font-size: 1.1em;
+  align-items: center;
+  height: 22px;
+`;
+
+const DateField = styled("span")`
+  font-weight: normal;
+  font-size: 0.8em;
+  color: ${colors.grey};
+`;
+
+const CommentsPanel = ({
+  match: {
+    params: { pageId },
+  },
+}) => {
+  const [comment, setComment] = useState("");
+  const { loading, error, data } = useQuery(COMMENT_QUERY, {
+    variables: {
+      input: { pageId },
+    },
+  });
+
+  const [createComment] = useMutation(COMMENT_ADD);
+
+  const handleChange = event => {
+    setComment(event.target.value);
+  };
+
+  const handleSubmit = event => {
+    if (comment) {
+      createComment({
+        variables: {
+          input: {
+            pageId,
+            commentText: comment,
+          },
+        },
+      });
+      setComment("");
+    }
+    event.preventDefault();
+  };
+
+  const getInitials = name => {
+    if (name !== null) {
+      const initials = name.replace(/[^a-zA-Z- ]/g, "").match(/\b\w/g);
+      return initials
+        .join("")
+        .substring(0, 3)
+        .toUpperCase();
+    }
+  };
+
+  if (loading) {
+    return <Loading height="100%">Comments loading…</Loading>;
+  }
+  if (error) {
+    return <Error>Oops! Something went wrong</Error>;
+  }
+
+  const commentsArray = get(data, "page.comments", []);
+
+  return (
+    <CommentsPane>
+      <ScrollPane>
+        <CommentSection>
+          <Field>
+            <Label>{"Comments"}</Label>
+            <TextArea
+              id="comments-txt-area"
+              name="textArea"
+              data-test="comment-txt-area"
+              value={comment}
+              onChange={handleChange}
+            />
+          </Field>
+          <ButtonGroup horizontal align="right">
+            <Button
+              disabled={!comment}
+              onClick={handleSubmit}
+              variant="primary"
+              data-test="btn-add-comment"
+            >
+              Add
+            </Button>
+          </ButtonGroup>
+        </CommentSection>
+
+        {commentsArray.map(item => (
+          <React.Fragment key={item.id}>
+            <CommentSection>
+              <CommentHeaderContainer>
+                <AvatarWrapper>
+                  <AvatarOuter>
+                    <AvatarInner>{getInitials(item.user.name)}</AvatarInner>
+                  </AvatarOuter>
+                </AvatarWrapper>
+                <NameWrapper>
+                  <FlexLabel>{item.user.name}</FlexLabel>
+                  <DateField>{moment(item.createdTime).calendar()}</DateField>
+                </NameWrapper>
+              </CommentHeaderContainer>
+              <ShowComment data-test="XYZ">{item.commentText}</ShowComment>
```

Comment:
Do we need this test attrib and if so could we make it a bit more descriptive?

Reply:
not used so removed